### PR TITLE
Fix grammar issues.

### PR DESCRIPTION
**Pull request by bangerth**
See #5539.

### CODE REVIEW

File: include/deal.II/base/index_set.h [hunk 1]
```diff
@@ -314,26 +314,26 @@ class IndexSet
...
+   * Remove all elements contained in @p other from this set. In other words,
    * if $x$ is the current object and $o$ the argument, then we compute $x
    * \leftarrow x \backslash o$.
    */
   void subtract_set (const IndexSet &other);
 
   /**
-   * Removes and returns the last element of the last range.
-   * Throws an exception if the IndexSet is empty.
+   * Remove and returns the last element of the last range.
```
Comment (drwells) [inline include/deal.II/base/index_set.h]: Should be 'remove and return'

File: include/deal.II/base/index_set.h [hunk 2]
```diff
@@ -314,26 +314,26 @@ class IndexSet
...
+   * This function throws an exception if the IndexSet is empty.
    */
   size_type pop_back ();
 
   /**
-   * Removes and returns the first element of the first range.
-   * Throws an exception if the IndexSet is empty.
+   * Remove and returns the first element of the first range.
```
Comment (drwells) [inline include/deal.II/base/index_set.h]: (same)

File: include/deal.II/base/partitioner.h [hunk 3]
```diff
@@ -561,13 +561,13 @@ namespace Utilities
       IndexSet ghost_indices_data;
 
       /**
-       * Caches the number of ghost indices. It would be expensive to use @p
+       * A variable that caches the number of ghost indices. It would be expensive to use @p
```
Comment (drwells) [inline include/deal.II/base/partitioner.h]: I think writing `A variable caching the number...` would sound better (same for `contains`, `stores`, `holds`, etc. below). What do you think?

Comment (bangerth) [reply include/deal.II/base/partitioner.h]: If you think so -- you're the native speaker, I'm not. I'm happy to defer to your sense of what sounds better.

**Comment by bangerth** [comment]
I made the requested changes.

/run-tests

**Pull request merged**